### DISCUSSION TITLE
Issues with editing custom fields

### DISCUSSION THREAD

**Discussion by dragn-dev** [swellstores/community#543] [Help] [answered]
Hello, I created a custom model called "Favorites" and I'm having issues editing it via the frontend client. I followed the steps to gain input field permissions but everytime I try to edit it I get the error that I don't have permissions to edit content.

These are my current permissions: 
![image](https://github.com/swellstores/community/assets/81596450/b87fbbcf-853b-449a-87cf-11cf5a4cbc13)

I tried both "favorites" and "Favorites".

This is the model:
![image](https://github.com/swellstores/community/assets/81596450/4d1978e2-fbdb-4341-ab7e-dc9a2166a938)

and these are the model data fields:
![image](https://github.com/swellstores/community/assets/81596450/90208f8f-b169-4d58-9a26-5ed9072e6bd1)

I'd appreciate any help because I'm not sure what I did wrong here.

**Comment by logeshswell** [comment, marked as answer]
> I tried both "favorites" and "Favorites".

"favorites" is correct.

1. Create a model "Favorites" without namespace.
<img width="1257" alt="Screenshot 2024-07-15 at 1 21 35 PM" src="https://github.com/user-attachments/assets/d4f41d9e-08b9-41ac-b164-1afe28422e8d">
2. Open up read/write permissions.

```json
{
  "permissions": {
    "favorites": {
      "fields": [
        "display"
      ],
      "input": {
        "fields": [
          "display"
        ]
      }
    }
  }
}
```
3. You can query and update data using swell-js as follows:

```js
await swell.get('/favorites');
await swell.put(`/favorites/${id}`, { display: 'value' });
```
